CollapseLauncher/XAMLs/MainApp/Pages/HomePage.xaml.cs

```diff
@@ -1836,14 +1839,14 @@ private async static void StartPlaytimeCounter(string oldRegionRegistryKey, Proc
                 {
                     while (true)
                     {
-                        Process[] pname = Process.GetProcessesByName(gamePreset.GameExecutableName.Split('.')[0]);
-                        switch (pname.Length)
+                        Process[] processName = Process.GetProcessesByName(gamePreset.GameExecutableName.Split('.')[0]);
+                        switch (processName.Length)
                         {
                             case 0:
                                 break;
                             case 1:
-                                proc = pname[0];
-                                LogWriteLine($"Found the main HI3 process [{pname[0].Id}]");
+                                proc = processName[0];
+                                LogWriteLine($"Found the main HI3 process [{processName[0].Id}]");
```

Comment:
I think these lines can be simplified to the code below:

```CSharp
if (gamePreset.GameType == GameType.Honkai)
{
    string processNameStr = Path.GetFileNameWithoutExtension(gamePreset.GameExecutableName);
    while (true)
    {
        // Try get the honkaiProc and auto-dispose it if the routine finishes,
        // then get the first process
        Process honkaiProc = Process.GetProcessesByName(processNameStr).FirstOrDefault();

        // If the honkaiProc is not null and also has the main window handle,
        // then continue to go through.
        // 
        // It replaces the switch statement where it checks for the .Length > 1
        // since the first process (process[0]) is not the actual active process.
        //
        // So, if it doesn't have one (IntPtr == Zero), then continue the loop until
        // the actual process found.
        if (honkaiProc != null && honkaiProc.MainWindowHandle != IntPtr.Zero)
        {
            // Assign the old proc with honkaiProc.
            proc = honkaiProc;

            // Wait the process
            LogWriteLine($"Found the main HI3 process [{proc.Id}]");
            await proc.WaitForExitAsync();
            break;
        }

        // If the process hasn't spawn yet or still has no MainWindowHandle,
        // then delay before continue to another loop routine.
        await Task.Delay(5000);
    }
}
// If not Honkai, then wait the process as usual
else await proc.WaitForExitAsync();
```

Reply:
welp, as per first comment, we need to use ``using`` statement for ``Process`` too. But since it's widely-used in the entire app, we may consider to make changes to it but that's for later discussion.

Reply:
Can we just make the process handler LinQ method used for GameBoost to public so it can be used here?

Reply:
> Can we just make the process handler LinQ method used for GameBoost to public so it can be used here?

I think we can make that after this PR is done

Reply:
> Can we just make the process handler LinQ method used for GameBoost to public so it can be used here?
Talking about GameBoost, I've seen now that that section of code already finds the actual main BH3 process so I think this section becomes redundant and the process found there can be used...

Reply:
As neon said above, we can use his suggestion for now and adjust it later when we do refactor/restructure before 1.73 merged to stable